src/crimson/os/seastore/seastore_types.cc

```diff
@@ -132,21 +132,21 @@ std::ostream &operator<<(std::ostream &out, const paddr_t &rhs)
 journal_seq_t journal_seq_t::add_offset(
       journal_type_t type,
       device_off_t off,
-      seastore_off_t roll_start,
-      seastore_off_t roll_size) const
+      device_off_t roll_start,
+      device_off_t roll_size) const
 {
   assert(offset.is_absolute());
   assert(off <= DEVICE_OFF_MAX && off >= DEVICE_OFF_MIN);
   assert(roll_start >= 0);
   assert(roll_size > 0);
 
   segment_seq_t jseq = segment_seq;
-  seastore_off_t joff;
+  device_off_t joff;
   if (type == journal_type_t::SEGMENTED) {
     joff = offset.as_seg_paddr().get_segment_off();
   } else {
     assert(type == journal_type_t::RANDOM_BLOCK);
-    auto boff = offset.as_blk_paddr().get_block_off();
+    auto boff = offset.as_blk_paddr().get_device_off();
     assert(boff <= MAX_SEG_OFF);
```

Comment:
assert(boff <= DEVICE_OFF_MAX) ?

Reply:
Dropped in the last commit. And boff from `blk_paddr_t::get_device_off()` is guaranteed to be reasonable from its constructor.